code_challenges/code_wars/morse_code.py

```diff
@@ -0,0 +1,74 @@
+def decode_morse(morse_code):
+    translator = {
+        "": "",
+        ".-": "A",
+        "-...": "B",
+        "-.-.": "C",
+        "-..": "D",
+        ".": "E",
+        "..-.": "F",
+        "--.": "G",
+        "....": "H",
+        "..": "I",
+        ".---": "J",
+        "-.-": "K",
+        ".-..": "L",
+        "--": "M",
+        "-.": "N",
+        "---": "O",
+        ".--.": "P",
+        "--.-": "Q",
+        ".-.": "R",
+        "...": "S",
+        "-": "T",
+        "..-": "U",
+        "...-": "V",
+        ".--": "W",
+        "-..-": "X",
+        "-.--": "Y",
+        "--..": "Z",
+        "-----": "0",
+        ".----": "1",
+        "..---": "2",
+        "...--": "3",
+        "....-": "4",
+        ".....": "5",
+        "-....": "6",
+        "--...": "7",
+        "---..": "8",
+        "----.": "9",
+        ".-.-.-": ".",
+        "--..--": ",",
+        "..--..": "?",
+        ".----.": "'",
+        "-.-.--": "!",
+        "-..-.": "/",
+        "-.--.": "(",
+        "-.--.-": ")",
+        ".-...": "&",
+        "---...": ":",
+        "-.-.-.": ";",
+        "-...-": "=",
+        ".-.-.": "+",
+        "-....-": "-",
+        "..--.-": "_",
+        ".-..-.": '"',
+        "...-..-": "$",
+        ".--.-.": "@",
+        "...---...": "SOS",
+    }
+    translation = ""
+    morse_code = morse_code.strip()
+    morse_code = morse_code.split(" ")
+
+    for i in range(len(morse_code)):
+        if morse_code[i] == "" and morse_code[i + 1] == "":
+            translation += " "
+        else:
+            translation += translator[morse_code[i]]
+    return translation
```

Comment:
This probably works to pass the tests.  Now let's think about a real world situation.  The morse code operator is tapping out a message.  What is the chance that in between each letter is exactly 3 spaces?  How would you code fail? How could you fix it?

Reply:
i guess they could accidentally put more than 3 spaces and with my code that would add an extra space to the translation. i guess i could add and and statement and say and morse_code[i + 2] != ""